src/com/xilinx/rapidwright/rwroute/PartialRouter.java

```diff
@@ -240,83 +240,26 @@ protected void routeGlobalClkNets() {
     protected void determineRoutingTargets() {
         super.determineRoutingTargets();
 
-        // Go through all nets to be routed
-        Map<RouteNode, RouteNode> stashedPrev = new HashMap<>();
-        for (Map.Entry<Net, NetWrapper> e : nets.entrySet()) {
-            Net net = e.getKey();
-            NetWrapper netWrapper = e.getValue();
-
-            // Temporarily stash the prev pointer of all sink nodes ahead of recovering routing
-            // (it's possible for another connection to use a bounce node, but now that node is
-            // needed as a site pin)
-            for (Connection connection : netWrapper.getConnections()) {
-                Consumer<RouteNode> action = (rnode) -> {
-                    if (rnode == null) {
-                        return;
-                    }
-                    RouteNode prev = rnode.getPrev();
-                    if (prev == null) {
-                        return;
-                    }
-                    stashedPrev.put(rnode, prev);
-                    rnode.clearPrev();
-                };
-                action.accept(connection.getSinkRnode());
-                connection.getAltSinkRnodes().forEach(action);
-            }
-
-            // Create all nodes used by this net and set its previous pointer so that:
-            // (a) the routing for each connection can be recovered by
-            //      finishRouteConnection()
-            // (b) RouteNode.setChildren() will know to only allow this incoming
-            //     arc on these nodes
-            for (PIP pip : net.getPIPs()) {
-                Node start = (pip.isReversed()) ? pip.getEndNode() : pip.getStartNode();
-                Node end = (pip.isReversed()) ? pip.getStartNode() : pip.getEndNode();
-
-                // Do not include arcs that the router wouldn't explore
-                // e.g. those that leave the INT tile, since we project pins to their INT tile
-                if (routingGraph.isExcludedTile(end))
-                    continue;
-
-                RouteNode rstart = getOrCreateRouteNode(start, null);
-                RouteNode rend = getOrCreateRouteNode(end, null);
-                assert (rend.getPrev() == null);
-                rend.setPrev(rstart);
-            }
-
-            // Use the prev pointers to attempt to recover routing for all indirect connections
-            for (Connection connection : netWrapper.getConnections()) {
-                if (connection.isDirect()) {
-                    continue;
-                }
-                RouteNode sourceRnode = connection.getSourceRnode();
-                RouteNode sinkRnode = connection.getSinkRnode();
-                assert(sourceRnode.getType() == RouteNodeType.PINFEED_O);
+        // With all routingGraph.preserveAsync() calls having completed,
+        // now check sink routability
```

Comment:
What does it mean to "check sink routability"?  Are we making sure there are no obvious blocking nodes to prevent node expansion out of the gate?